### PR TITLE
Sjq2t-17_authentication_be

### PR DESCRIPTION
**Pull request by DarkoVeljanoskiHtec**
Everything is set up for the authentication, I've added two new filters for the authentication and authorization and made changes to the web security class, register and login works.
The get users functionality is not implemented, I'll create new task on Jira for that.

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/SkyNest/model/request/UserLoginRequest.java [hunk 1]
```diff
@@ -1,8 +1,12 @@
 package com.htecgroup.SkyNest.model.request;
 
 import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
 
 @Data
+@Setter
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/SkyNest/model/request/UserLoginRequest.java]: Doesn't Data already make all Getters and Setters?

Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/SkyNest/model/request/UserLoginRequest.java]: I think you're right, I didn't know that.

File: backend/src/main/java/com/htecgroup/SkyNest/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -51,4 +51,6 @@ public UserDto findUserByEmail(String email) {
 
     return modelMapper.map(userEntity, UserDto.class);
   }
+
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/SkyNest/service/impl/UserServiceImpl.java]: Try to format or check diffs before pushing, so we don't get changes like this :) .

Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/SkyNest/service/impl/UserServiceImpl.java]: That's good suggestion, I'll check it from now.

File: backend/src/main/java/com/htecgroup/SkyNest/filter/CustomAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                        .signWith(SignatureAlgorithm.HS512, SecurityConstants.TOKEN_SECRET)
+                        .compact();
+
+        UserService userService = (UserService) SpringApplicationContext.getBean("userServiceImpl");
+        UserDto userDto = userService.findUserByEmail(userName);
+
+        response.addHeader(SecurityConstants.HEADER_STRING, SecurityConstants.TOKEN_PREFIX + token);
+
+        response.addHeader("UserID", userDto.getId().toString());
+
```
Comment (DraganMitrasinovicHtec) [inline backend/src/main/java/com/htecgroup/SkyNest/filter/CustomAuthenticationFilter.java]: Was it intended to have this line twice?

Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/SkyNest/filter/CustomAuthenticationFilter.java]: it was mistake, I changed it.

**Pull request closed without merging**